### ISSUE TITLE
Invalid transaction amount, -XYZ

### ISSUE THREAD

**Issue by Shekelme** [iron-fish/ironfish#2532]
### What happened?

A bug happened! The execution of the `ironfish depositAll` command is interrupted with an error `Invalid transaction amount, -980000000` (the value may vary), this can happen several times a day.

### Version

v0.1.51

### Debug output

```shell
Iron Fish version                  0.1.51 @ 67da183
Iron Fish library                  0.0.28 @ 67da183
Operating system                   Linux x64
CPU model(s)                       AMD Ryzen Threadripper PRO 3955WX 16-Cores
CPU threads                        32
RAM total                          503.60 GiB
Heap total                         32.00 GiB
Node version                       v16.17.0
ironfish in PATH                   true
Garbage Collector Exposed          false
Telemetry enabled                  true
Node name                          len16
Block graffiti                     saturnia
Account d3a8a9 uuid                d3a8a9c2-aebe-4669-95c3-cb63c9eb88b6
Account d3a8a9 name                default
Account d3a8a9 head hash           0000000000001f437f96636fb8cf05b0a88de8c8793a0d815b69e576c4b4d68e
Account d3a8a9 head in chain       true
Account d3a8a9 sequence            264582
```

### Graffiti

saturnia

**Comment by ADADarkpool** [comment]
I have this issue as well.

**Comment by hughy** [comment]
This is happening because of network forks. depositAll is creating transactions that spend notes from forks, but once the node has reorganized from the fork the account appears to have spent notes that don't exist, so the balance turns negative until the invalid transaction either becomes valid or expires.

We're putting in a fix so that the invalid transactions aren't included in the balance.

**Comment by hughy** [comment]
Resolved by #2565
